### PR TITLE
[CHORE]API Requirements made by elsayeed

### PR DESCRIPTION
**Pull request by NoBoneZ**
## Is this a Feature, Chore or Bug?
- [] Feature
- [x] Chore
- [] Bug

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/api/v1/authentication.py [hunk 1]
```diff
@@ -351,3 +357,57 @@ def fetch_employee_checkin_list(from_date=None, to_date=None, limit=20, page_num
 		"number of checkin": len(check_list)
 	}
 	return response("Successfully Retrieved !", 200, data)
+
+@frappe.whitelist(allow_guest=True)
+def new_forgot_password(employee_id=None):
```
Comment (pjamsheer) [inline one_fm/api/v1/authentication.py:373]: What is the use of this method

File: one_fm/api/v1/authentication.py [hunk 2]
```diff
@@ -285,12 +285,18 @@ def send_token_via_email(user, token, otp_secret, otp_issuer, subject=None, mess
...
 		return response("Employee ID cannot be None", 401, None, "Employee ID is required !")
-	doc = frappe.db.get_value("Employee",{ "employee_id": employee_id}, "employee_name")
-	if doc == None:
+	doc = frappe.get_doc("Employee",{ "employee_id": employee_id})
+	if not doc:
 		return response("Employee Not Found", 404, None, "Employee ID of an active Employee is required")
-	return response("Success", 200, doc)
-
-
+	registration_status = frappe.db.get_value("User", doc.user_id, "last_password_reset_date")
```
Comment (pjamsheer) [inline one_fm/api/v1/authentication.py:301]: @NoBoneZ Add inline comments in the API arguments please

File: one_fm/api/v1/authentication.py [hunk 3]
```diff
@@ -285,12 +285,18 @@ def send_token_via_email(user, token, otp_secret, otp_issuer, subject=None, mess
...
+	client_id = frappe.db.get_value("OAuth Client", {"app_name": "OneFM" }, "client_id")
+	data = {
+		"name in arabic": doc.employee_name_in_arabic,
+		"name in english": doc.employee_name,
+		"is_registered": True if registration_status else False,
+		"client_id": client_id
```
Comment (pjamsheer) [inline one_fm/api/v1/authentication.py]: Add is_entrolled data too with this response

File: one_fm/api/v1/authentication.py (hunk 1 above)
Comment (NoBoneZ) [reply one_fm/api/v1/authentication.py:373]: This checks if the employee exists, and returns the user_id, if it exists...The user_id is important as it is used to get the otp

File: one_fm/api/v1/authentication.py (hunk 2 above)
Comment (NoBoneZ) [reply one_fm/api/v1/authentication.py:301]: Done

File: one_fm/api/v1/authentication.py (hunk 3 above)
Comment (NoBoneZ) [reply one_fm/api/v1/authentication.py]: Included

**Review by pjamsheer — APPROVED** [review]

**Pull request merged**